planner/core/optimizer.go

```diff
@@ -83,9 +84,25 @@ var optRuleList = []logicalOptRule{
 	&columnPruner{}, // column pruning again at last, note it will mess up the results of buildKeySolver
 }
 
+type logicalOptimizeOp struct {
+	enableTrace bool
+}
+
+type logicalOptimizeOption func(*logicalOptimizeOp)
+
+func withEnableOptimizeTrace() logicalOptimizeOption {
+	return func(op *logicalOptimizeOp) { op.enableTrace = true }
+}
+
+type logicalOptimizeReturn struct {
+	logicalPlan LogicalPlan
+	// comment it for now in order to pass lint, we will use steps in future
+	// optimizeTraceSteps []tracing.LogicalRuleOptimizeTraceStep
+}
+
 // logicalOptRule means a logical optimizing rule, which contains decorrelate, ppd, column pruning, etc.
 type logicalOptRule interface {
-	optimize(context.Context, LogicalPlan) (LogicalPlan, error)
+	optimize(context.Context, LogicalPlan, ...logicalOptimizeOption) (*logicalOptimizeReturn, error)
```

Comment:
Please add one rule implementation to show how this framework works.

Reply:
updated in `aggregationEliminateChecker`.